beacon_node/beacon_chain/src/canonical_head.rs

```diff
@@ -447,6 +456,12 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         self.canonical_head.cached_head_read_lock().snapshot.clone()
     }
 
+    /// Returns the `AnchorState` of the finalized checkpoint
+    /// TODO: determine if this puts a burdon on the fork choice lock
```

Comment:
So this is used every time we add a peer who's finalized is lower than ours. Might not be a big deal in practice. But if it is, we could track "has our anchor state ever been set to `AnchorState::Finalized`" because it will never go from `AnchorState::NonRevertible` => `AnchorState::Finalized`. If we track that, we won't need to bother fork choice as soon as finalization advances at any point after startup